app/src/main/java/zapsolutions/zap/connection/manageWalletConfigs/WalletConfigsManager.java

```diff
@@ -0,0 +1,219 @@
+
+import zapsolutions.zap.baseClasses.App;
+import zapsolutions.zap.util.PrefsUtil;
+
+/**
+ * This SINGLETON class is used to load and save configurations for wallets.
+ * Multiple wallets can exist simultaneously, but each alias (wallet name) is only allowed to exist once.
+ * <p>
+ * The wallet configurations are stored encrypted in the default shared preferences.
+ */
+public class WalletConfigsManager {
+
+    private static final String LOG_TAG = WalletConfigsManager.class.getName();
+
+    public static final String DEFAULT_WALLET_NAME = "DefaultWallet";
+
+    private static WalletConfigsManager mInstance;
+    private WalletConfigsJson mWalletConfigsJson;
+
+    private WalletConfigsManager() {
+
+        String encrypted = PrefsUtil.getPrefs().getString(PrefsUtil.WALLET_CONFIGS, "");
+
+        String decrypted = null;
+        try {
+            decrypted = new Cryptography(App.getAppContext()).decryptData(encrypted);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        try {
+            mWalletConfigsJson = new Gson().fromJson(decrypted, WalletConfigsJson.class);
+        } catch (JsonSyntaxException e) {
+            createEmptyWalletConfigsJson();
+        }
+        if (mWalletConfigsJson == null) {
+            createEmptyWalletConfigsJson();
+        }
+    }
+
+    // used for unit tests
+    public WalletConfigsManager(String walletConfigsJson) {
+        try {
+            mWalletConfigsJson = new Gson().fromJson(walletConfigsJson, WalletConfigsJson.class);
+        } catch (JsonSyntaxException e) {
+            createEmptyWalletConfigsJson();
+        }
+        if (mWalletConfigsJson == null) {
+            createEmptyWalletConfigsJson();
+        }
+    }
+
+    public WalletConfigsJson getWalletConfigsJson() {
+        return mWalletConfigsJson;
+    }
+
+    public static WalletConfigsManager getInstance() {
+        if (mInstance == null) {
+            mInstance = new WalletConfigsManager();
+        }
+
+        return mInstance;
+    }
+
+    private void createEmptyWalletConfigsJson() {
+        mWalletConfigsJson = new Gson().fromJson("{\"connections\":[]}", WalletConfigsJson.class);
+    }
+
+
+    /**
+     * Used to determine if the provided String is a valid walletConfigs JSON.
+     *
+     * @param walletConfigsString parses as JSON
+     * @return if the JSON syntax is valid
+     */
+    public static boolean isValidJson(String walletConfigsString) {
+        try {
+            WalletConfigsJson walletConfigs = new Gson().fromJson(walletConfigsString, WalletConfigsJson.class);
+            return walletConfigs != null;
+        } catch (JsonSyntaxException ex) {
+            return false;
+        }
+    }
+
+
+    /**
+     * Checks if a wallet configuration with the given alias/name already exists.
+     *
+     * @param alias The name of the wallet
+     * @return
+     */
+    public boolean doesWalletConfigExist(@NonNull String alias) {
+
```

Comment:
Would remove the empty line.

We can use `mWalletConfigsJson.doesWalletConfigExist(alias);`